examples/nodejs/src/marlowe-object-flow.ts

```diff
@@ -0,0 +1,659 @@
+ * - How to stake the assets of a contract to a given stake address
+ * - How to validate that a Merkleized contract is an instance of a given contract
+ * - How to share contract sources between different runtimes
+ */
+import { mkLucidWallet, WalletAPI } from "@marlowe.io/wallet";
+import { mkRuntimeLifecycle } from "@marlowe.io/runtime-lifecycle";
+import { Lucid, Blockfrost, C } from "lucid-cardano";
+import { readConfig } from "./config.js";
+import { datetoTimeout } from "@marlowe.io/language-core-v1";
+import {
+  contractId,
+  ContractId,
+  contractIdToTxId,
+  stakeAddressBech32,
+  StakeAddressBech32,
+  Tags,
+  transactionWitnessSetTextEnvelope,
+  TxId,
+} from "@marlowe.io/runtime-core";
+import { Address } from "@marlowe.io/language-core-v1";
+import { ContractBundle, lovelace } from "@marlowe.io/marlowe-object";
+import { input, select } from "@inquirer/prompts";
+import { RuntimeLifecycle } from "@marlowe.io/runtime-lifecycle/api";
+import {
+  AppliedActionResult,
+  getApplicableActions,
+  mkApplicableActionsFilter,
+} from "./experimental-features/applicable-inputs.js";
+import arg from "arg";
+import { splitAddress } from "./experimental-features/metadata.js";
+import { SingleInputTx } from "../../../packages/language/core/v1/dist/esm/transaction.js";
+import * as t from "io-ts/lib/index.js";
+import { deepEqual } from "@marlowe.io/adapter/deep-equal";
+
+// When this script is called, start with main.
+main();
+
+// #region Command line arguments
+function parseCli() {
+  const args = arg({
+    "--help": Boolean,
+    "--config": String,
+    "-c": "--config",
+  });
+
+  if (args["--help"]) {
+    printHelp(0);
+  }
+  function printHelp(exitStatus: number): never {
+    console.log("Usage: npm run marlowe-object-flow -- --config <config-file>");
+    console.log("");
+    console.log("Example:");
+    console.log("  npm run marlowe-object-flow -- --config alice.config");
+    console.log("Options:");
+    console.log("  --help: Print this message");
+    console.log(
+      "  --config | -c: The path to the config file [default .config.json]"
+    );
+    process.exit(exitStatus);
+  }
+  return args;
+}
+
+// #endregion
+
+// #region Interactive menu
+
+/**
+ * Small command line utility that prints a confirmation message and writes dots until the transaction is confirmed
+ * NOTE: If we make more node.js cli tools, we should move this to a common place
+ */
+async function waitIndicator(wallet: WalletAPI, txId: TxId) {
+  process.stdout.write("Waiting for the transaction to be confirmed...");
+  const intervalId = setInterval(() => {
+    process.stdout.write(".");
+  }, 1000);
+  await wallet.waitConfirmation(txId);
+  clearInterval(intervalId);
+  process.stdout.write("\n");
+}
+
+/**
+ * This is an Inquirer.js validator for bech32 addresses
+ * @returns true if the address is valid, or a string with the error message otherwise
+ */
+function bech32Validator(value: string) {
```

Comment:
We should add this kind of behavior in our branded types eventually...

Reply:
yeap, ideally not relying on a 4mb library